internal/headers.go

```diff
@@ -36,8 +36,25 @@ type HeaderReader interface {
 	ForEachKey(handler func(string, []byte) error) error
 }
 
-// ContextPropagator is an interface that determines what information from
-// context to pass along
+// ContextPropagator determines what information from context to pass along.
```

Comment:
thanks for filling in this documentation gap. @shijiesheng can you review please?